src/Lean/LocalContext.lean

```diff
@@ -15,7 +15,14 @@ Each declaration comes with
 - `fvarId` the unique id of the free variables
 - `userName` the pretty-printable name of the variable
 - `type` the type.
-A `cdecl` is a local variable, a `ldecl` is a let-bound free variable with a `value : Expr`.
+A `cdecl` is a local variable (the `c` stands for 'c'onstant declaration for legacy reasons) with no value,
+and an `ldecl` is a let-bound free variable with a `value : Expr`.
+
+The flag `nonDep` on an `ldecl` tracks whether `let x := v; b` is equivalent to `(fun x => b) v`.
+This is always equal if the let-binding is non-dependent. An example where these are inequivalent is
+the case where
+`let n : Nat := 2; fun (a : Array Nat n) (b : Array Nat 2) => a = b` is type correct, but
+`(fun (n : Nat) (a : Array Nat n) (b : Array Nat 2) => a = b) 2` is not.
```

Comment:
I think we should refer to `Expr.letE`'s docstring here instead of rehashing it.

Reply:
@bollu Did you have the opportunity to address this comment from @Kha?
Thanks, Leo